lib/actions/field-trip.js

```diff
@@ -149,192 +234,114 @@ export function addFieldTripNote (request, note, intl) {
 }
 
 /**
- * Delete a specific note for a field trip request.
+ * Invokes OTP Datatstore to update a field trip request or its related notes or itineraries.
  */
-export function deleteFieldTripNote (request, noteId, intl) {
+function updateFieldTripRequest(
+  request,
+  endpoint,
+  data,
+  intl,
+  getErrorMessage,
+  includeRequestId
+) {
   return function (dispatch, getState) {
-    const {callTaker, otp} = getState()
-    const {datastoreUrl} = otp.config
+    const { callTaker, otp } = getState()
+    const { datastoreUrl } = otp.config
     if (sessionIsInvalid(callTaker.session)) return
-    const {sessionId} = callTaker.session
-    return fetch(`${datastoreUrl}/fieldtrip/deleteNote`,
-      {body: serialize({ noteId, sessionId }), method: 'POST'}
-    )
+    const { sessionId } = callTaker.session
+    const payload = {
+      sessionId,
+      ...data
+    }
+    if (includeRequestId) {
+      payload.requestId = request.id
+    }
+    return fetch(`${datastoreUrl}/fieldtrip/${endpoint}`, {
+      body: serialize(payload),
+      method: 'POST'
+    })
       .then(() => dispatch(fetchFieldTripDetails(request.id, intl)))
-      .catch(err => {
-        alert(
-          intl.formatMessage(
-            { id: 'actions.fieldTrip.deleteNoteError' },
-            { err: JSON.stringify(err) }
-          )
-        )
-      })
+      .catch((err) => alert(`${getErrorMessage()} ${JSON.stringify(err)}`))
   }
 }
 
 /**
- * Edit teacher (AKA submitter) notes for a field trip request.
+ * Delete a specific note for a field trip request.
  */
-export function editSubmitterNotes (request, submitterNotes, intl) {
-  return function (dispatch, getState) {
-    const {callTaker, otp} = getState()
-    const {datastoreUrl} = otp.config
-    if (sessionIsInvalid(callTaker.session)) return
-    const {sessionId} = callTaker.session
-    const noteData = serialize({
-      notes: submitterNotes,
-      requestId: request.id,
-      sessionId
-    })
-    return fetch(`${datastoreUrl}/fieldtrip/editSubmitterNotes`,
-      {body: noteData, method: 'POST'}
-    )
-      .then(() => dispatch(fetchFieldTripDetails(request.id, intl)))
-      .catch(err => {
-        alert(
-          intl.formatMessage(
-            { id: 'actions.fieldTrip.editSubmitterNotesError' },
-            { err: JSON.stringify(err) }
-          )
-        )
-      })
-  }
+export function deleteFieldTripNote(request, noteId, intl) {
+  return updateFieldTripRequest(
+    request,
+    'deleteNote',
```

Comment:
Do you think it might be a good idea to make these an enum to avoid room for error when editing this in the future

Reply:
Handled in https://github.com/opentripplanner/otp-react-redux/pull/627/commits/42a8fb2c0864378234354ceb9827ddf380ac03bc.